sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -1976,9 +1976,9 @@ public async Task CreatePartitionProcessorProcessingTaskStartsTheConsumerAtTheCo
         public async Task CreatePartitionProcessorProcessingTaskDoesNotReplaceTheConsumerOnFatalExceptions()
         {
             using var cancellationSource = new CancellationTokenSource();
-            cancellationSource.CancelAfter(TimeSpan.FromSeconds(15));
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(20));
 
-            var expectedException = new StackOverflowException("We be overflowing!");
```

Comment:
This was being treated inconsistently across target platforms with Linux having a good deal of non-determinism around creating and bubbling, often exceeding the timeout.  As this and most of the other known-fatal exceptions are typically runtime generated, moving to the `TaskCanceledException` was the safest path.   In order to ensure that the correct exception was observed, a customized message is used to differentiate.